syntax = "proto3";

package Acts.protobuf;

message MaterialMap {
  
  message Dimension {
    uint32 nBins = 1;
    double min = 2;
    double max = 3;
  }

  Dimension l0 = 1;
  Dimension l1 = 2;

  uint64 geo_id = 3;

  int32 vol_id = 4;
  int32 lay_id = 5;
  int32 app_id = 6;
  int32 sen_id = 7;

  message MaterialProperties {
    double thickness = 1;
    double X0 = 2;
    double L0 = 3;
    double A = 4;
    double Z = 5;
    double rho = 6;
  }

  repeated MaterialProperties bins = 8;
}
